src/gui/previewselectdialog.cpp

```diff
@@ -135,3 +142,35 @@ void PreviewSelect::previewButtonClicked()
 
     accept();
 }
+
+void PreviewSelectDialog::saveSettings()
+{
+    // Persist modal size
+    m_storeDialogSize = this->size();
+    // Persist TreeView Header state
+    m_storeTreeHeaderState = previewList->header()->saveState();
+}
+
+void PreviewSelectDialog::loadSettings()
+{
+    // Restore modal size
+    if (m_storeDialogSize.value().isValid()) {
+        resize(m_storeDialogSize);
+    }
+    // Restore TreeView Header state
+    if (!m_storeTreeHeaderState.value().isEmpty()) {
+        m_headerStateRestored = previewList->header()->restoreState(m_storeTreeHeaderState);
+    }
+}
+
+void PreviewSelectDialog::showEvent(QShowEvent *event)
+{
+    Q_UNUSED(event);
+
+    // Default size, have to be called after show(), because width is needed
+    // Set Name column width to 60% of TreeView
+    if (!m_headerStateRestored) {
+        int nameSize = (previewList->size().width() * 0.6);
+        previewList->header()->resizeSection(0, nameSize);
```

Comment:
`m_headerStateRestored = true;` ?

Reply:
The value is filled by previewList->header()->restoreState(m_storeTreeHeaderState); https://github.com/qbittorrent/qBittorrent/pull/7429/files#diff-e9b7c9e34128ec8f30010cd8194835d9R162 because of that i named it m_headerStateRestored, did you see that, or should i rename?

Reply:
1. The renaming is just a suggestion, you can decide it.
2. I did saw the code you mentioned, however I also saw from Qt doc: http://doc.qt.io/qt-5/qshowevent.html#details
  > There are two kinds of show events

  So I wonder if you should set it to true explicitly to avoid the second redundant execution.

Reply:
Initial `false` value is needed, when there is no value saved in preferences, I think it's good like its now.

Reply:
No, I meant:
```c++
if (!m_headerStateRestored) {
  int nameSize = (previewList->size().width() * 0.6);
  previewList->header()->resizeSection(0, nameSize);
  m_headerStateRestored = true;  // maybe add this line
}
```

Reply:
Now I really tested it, my above comment is right, it is to avoid resizing header constantly when I minimize & restore the dialog on first launch.

Reply:
You are right, I didn't understand before, I though that you mean the init statement, I push changes today

Reply:
Done @Chocobo1